### PR TITLE
Create `PrintDiamondPattern.js`

### LINKED ISSUE #4674: Write a JavaScript program to print diamond pattern
### Description

Write a JavaScript program to print diamond pattern

```
Input  : 5

Output : 
    *
   ***
  *****
 *******
*********
 *******
  *****
   ***
    *
```
<details>
<summary>How to contribute</summary>

- Save the solution in `program/PrintDiamondPattern.js` file
- Add `PrintDiamondPattern.js` file in `print-diamond-pattern` folder

</details>

### PR DESCRIPTION
**Pull request by anandfresh**
Closes codinasion/codinasion#3547

### CODE REVIEW

**Review by oyetanishq — CHANGES_REQUESTED** [review]
please mention the issue number with it

File: program/print-diamond-pattern/PrintDiamondPattern.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+   const construct = ['*'.repeat(n)];
+   while((n-=2) > 0) {
+      const layer = ' '.repeat(++iteration)+'*'.repeat(n);
+      construct.unshift(layer); 
+      construct.push(layer);   
+   }
+   construct.forEach((pattern) => console.log(pattern)); 
+}
+
+diamondPattern(11);
```
Comment (joao-vitor-souza) [inline program/print-diamond-pattern/PrintDiamondPattern.js:12]:
```suggestion
diamondPattern(5);
```

File: program/print-diamond-pattern/PrintDiamondPattern.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const diamondPattern = (n) => {
+   let total = n, iteration = 0;
+   const construct = ['*'.repeat(n)];
+   while((n-=2) > 0) {
+      const layer = ' '.repeat(++iteration)+'*'.repeat(n);
```
Comment (joao-vitor-souza) [inline program/print-diamond-pattern/PrintDiamondPattern.js:5]: Change to this, so it'll work as shown in #4674

```suggestion
   let iteration = 0;
   const construct = ['*'.repeat(2 * n - 1)];
   n *= 2;
   while ((n -= 2) > 0) {
      const layer = ' '.repeat(++iteration) + '*'.repeat(n - 1);
```

**Review by joao-vitor-souza — CHANGES_REQUESTED** [review]

**Comment by joao-vitor-souza** [comment]
I've linked the issue for you, but you need to assign it to yourself. To do this you have to comment `!assign` in codinasion/codinasion#3547

**Comment by anandfresh** [comment]
issue: https://github.com/codinasion/codinasion/issues/3547

**Review by oyetanishq — APPROVED** [review]

**Review by joao-vitor-souza — APPROVED** [review]

**Pull request merged**